server.js

```diff
@@ -1,38 +1,167 @@
 mongoose.Promise = Promise
 
+//HOUSE MODEL
+const House = mongoose.model('House', {
+  name: String,
+  mascot: String,
+  head_teacher: String,
+  ghost: String,
+  founder: String
+})
+
+// const Book = mongoose.model('Book', {
+//   title: String,
+//   house: {
+//     type: mongoose.Schema.Types.ObjectId,
+//     ref: 'House'
+//   }
+// })
+
+//POTTER CHARACTER MODEL
+const Character = mongoose.model('Character', {
+  id: Number,
+  name: String,
+  gender: String,
+  job: String,
+  house: String,
+  wand: String,
+  patronus: String,
+  species: String,
+  blood_status: String,
+  hair_colour: String,
+  eye_colour: String,
+  loyalty: String,
+  skills: String,
+  birth: String,
+  death: String
+})
+
+//Wrap the seed in an enviorment variable to prevent it from re-run everytime we start the server. RESET_DB=true npm run dev
+if (process.env.RESET_DB) {
+  console.log('Resetting database!')
+  const seedDatabase = async () => {
+    await Character.deleteMany()
+    await House.deleteMany()
+    // await Book.deleteMany()
+
+    const gryffindor = new House({ name: "Gryffindor", mascot: "Lion", head_teacher: "Minverva McGonogall", ghost: "Nearly Headless Nick", founder: "Godric Gryffindor" })
+    await gryffindor.save()
+
+    const slyterhin = new House({ name: "Slytherin", mascot: "Snake", head_teacher: "Severus Snape", ghost: "The Bloody Baron", founder: "Salazar Slytherin" })
+    await slyterhin.save()
+
+    const ravenclaw = new House({ name: "Ravenclaw", mascot: "Eagle", head_teacher: "Filius Flitwick", ghost: "The Grey Lady", founder: "Rowena Ravenclaw" })
+    await ravenclaw.save()
+
+    const hufflepuff = new House({ name: "Hufflepuff", mascot: "Badger", head_teacher: "Pomona Sprout", ghost: "the Far Friar", founder: "Helga Hufflepuff" })
+    await hufflepuff.save()
+
+    // await new Book({ title: "Harry Potter and the Philosopher's Stone", house: gryffindor }).save()
+    // await new Book({ title: "Harry Potter and the Chamber of Secrets", house: slyterhin }).save()
+
+    potterData.forEach((character) => {
+      new Character(character).save()
+    })
+  }
+  seedDatabase()
+}
+
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
-//
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 9090
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// MIDDLEWARES
 app.use(cors())
 app.use(bodyParser.json())
+//A function that recives a request, the response and the argument "next". 
+//It will execute before the routes below, if it's not evoked next() it will block the code coming next. The connection.readyState checks that the connection is good.
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
```

Comment:
Interesting! This essentially checks if mongoose can establish a connection before trying to respond to any request. Hooray for custom middleware!